### PR TITLE
fix:deprecation removed and hsm added

### PR DESCRIPTION
**Pull request by d4kverma**
## what
* deprecation removed 
* hsm resources added. 
* github action added and updated old one.

### CODE REVIEW

**Review by 13archit — COMMENTED** [review]
In ACR for private endpoint, a records are automatically created in private dns and thus we don't require to get private ip for private endpoint with data block, is not for key_vault?

File: main.tf [hunk 1]
```diff
@@ -200,86 +331,31 @@ resource "azurerm_private_dns_a_record" "arecord-1" {
...
-    "verify",
-    "wrapKey",
-  ]
-  tags = module.labels.tags
-}
-
+##----------------------------------------------------------------------------- 
+## Below resources will create diagnostic setting for key vault and its components. 
+##-----------------------------------------------------------------------------
 resource "azurerm_monitor_diagnostic_setting" "example" {
```
Comment (13archit) [inline main.tf:337]: can we change use meaningful reference rather than using `example`

File: main.tf [hunk 2]
```diff
@@ -290,21 +366,45 @@ resource "azurerm_monitor_diagnostic_setting" "pe_kv_nic" {
...
   eventhub_authorization_rule_id = var.eventhub_authorization_rule_id
   log_analytics_workspace_id     = var.log_analytics_workspace_id
   log_analytics_destination_type = var.log_analytics_destination_type
-  metric {
-    category = "AllMetrics"
-    enabled  = var.Metric_enable
-    retention_policy {
-      enabled = var.retention_policy_enabled
-      days    = var.days
+  dynamic "metric" {
```
Comment (13archit) [inline main.tf:375]: Don't we require `enabled_log` for this?

Comment (d4kverma) [reply main.tf:375]: No . its not support for network interface card diagnosis setting.

File: main.tf (hunk 1 above)
Comment (d4kverma) [reply main.tf:337]: we cannot provide a details about each resource. Like diagnosis setting is used for reason and other, if someone is using any resource he/she should has idea about diagnosis setting.

**Pull request merged**